### PR TITLE
Chokidar for watching

### PR DESCRIPTION
**Pull request by jeremiak**
This is related to #2563 but it mostly is aimed at getting the site to build, and build on changes, in a reasonable amount of time on @brentryanjohnson's machine. The major feature of this pull request is to use node (and a library called [`chokidar`](https://npm.im/chokidar)) to watch for file changes and then tell Jekyll to build the site instead of relying on Jekyll's watch feature. In our testing, it seemed that Jekyll, especially when run inside of Docker, would frequently miss file changes and provide no indication of if it had noticed changes and was processing them.

With our new configuration, we wrote more code to do the watching, but it is more reliable and enables us to output information to the console so that site developers are made aware of the system recognizing changes.

Because we are already using `docker-compose` I hope that this change is mostly invisible to developers and should work with `docker-compose up`.

### CODE REVIEW

**Comment by coreycaitlin** [comment]
@jeremiak This didn't break anything for me, but I'm not sure it fixed it, either! I'm still able to run `docker-compose up` successfully, and that does build the site. But I'm not seeing any new indication that it's seeing and regenerating the site based on changes to markdown files. Happy to pair on this if it'd be helpful for you to see what I'm seeing!

**Comment by jeremiak** [comment]
@coreycaitlin Let's pair tomorrow on this! Maybe before you do your docker onboarding session with @alexpandel so that we can make sure you have at least one working implementation?

File: docker-compose.yml [hunk 1]
```diff
@@ -1,20 +1,35 @@
 version: '2'
 services:
-  app:
+  chokidar:
```
Comment (msecret) [inline docker-compose.yml]: Is there a reason we should call it this rather then just `app`?

**Comment by coreycaitlin** [comment]
Okay, I've confirmed that this _does_ fix the "build twice to make styles show up" problem 🎉

File: docker-compose.yml (hunk 1 above)
Comment (jeremiak) [reply docker-compose.yml]: I'm open to just calling this `app`, I don't have a really strong opinion. My thinking was that it is a bit nicer in the output to actually know what the service was. Maybe we should call this `jekyll`?

**Comment by coreycaitlin** [comment]
We've confirmed that watching is _working_ 👍 but chokidar is not watching enough files (currently only watching files in the `_layout` directory. Once we address that, let's get this merged in!

File: scripts/jekyll-watch.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        resolve();
+      } else {
+        reject(`${cmdline} exited with code ${code}!`);
+      }
+    });
+  });
+}
+
+function runJekyll() {
+  return run('/doi', 'jekyll', ['build', '--incremental'])
```
Comment (msecret) [inline scripts/jekyll-watch.js]: Is any of the stuff happening in `serve-fast` still relevant here (besides `--incremental`)?

Comment (jeremiak) [reply scripts/jekyll-watch.js]: I don't believe so. I'd like to remove the many different `*serve*` files and only have on Jekyll config. The host binding and such happens with docker and I don't know that we want to run the profiler/tracer every time we run the build. What do you think?

Comment (msecret) [reply scripts/jekyll-watch.js]: Agreed

Comment (jeremiak) [reply scripts/jekyll-watch.js]: Done in https://github.com/18F/doi-extractives-data/pull/2613

**Comment by coreycaitlin** [comment]
We've confirmed that this PR successfully resolves build issues so that:

1. The site builds all files successfully, in the right order, on the first try
2. It watches for file changes and rebuilds the site when file changes are saved
3. Icon fonts are appearing correctly in local builds

The Federalist issue is *not* related to any changes made here, but is an unrelated (albeit annoying) Federalist/cloud.gov issue.

Let's merge this puppy 🐶 🚀

**Comment by jeremiak** [comment]
Right now this is failing to build on Federalist but that is actually an issue with the platform and not this project. The cloud.gov team recently updated their platform to ensure that containers can only use 2GB of disk space, and between this projects, its dependencies, and Node, Ruby, and Python (for Federalist) the project weighed in at 2.4GB. Thus, Federalist isn't able to finish our builds.

The Federalist team is working on expanding the container size to 4GB, but @coreycaitlin and @msecret have tested this on their machines so we're going to merge now.

**Pull request merged**